prereise/gather/demanddata/bldg_electrification/load_projection.py

```diff
@@ -0,0 +1,674 @@
+
+
+def scale_energy(
+    base_energy, temp_df, base_scen, new_scen, midperfhp_cop, advperfhp_cop
+):
+    """project energy consumption for each projection scenarios from the base scenario
+
+    :param pandas.DataFrame base_energy: dataframe of disaggregated electricity consumptions for all weather years
+    :param pandas.DataFrame temp_df: weather records the given hours
+
+    :return (*pandas.DataFrame*) scen_load_mwh -- hourly electricity consumption induced by heat pump heating, resistance heating, cooling, and 'base' loads for a projection scenario
+    """
+    base_load_scaler = new_scen.floor_area_growth(base_scen)
+    cool_load_scaler = new_scen.frac_cooling_eff_change(
+        base_scen
+    ) * new_scen.frac_cool_growth(base_scen)
+    resist_load_scaler = new_scen.frac_resist_growth(base_scen)
+    if new_scen.compare_hp_heat_type(base_scen):
+        hp_load_scaler = new_scen.frac_hp_growth(base_scen)
+    else:
+        if base_scen.hp_type_heat == "midperfhp":
+            base_hp_heat_cop = midperfhp_cop
+        elif base_scen.hp_type_heat == "advperfhp":
+            base_hp_heat_cop = advperfhp_cop
+        if new_scen.hp_type_heat == "midperfhp":
+            new_hp_heat_cop = midperfhp_cop
+        elif new_scen.hp_type_heat == "advperfhp":
+            new_hp_heat_cop = advperfhp_cop
+
+        hp_cop_adv = new_hp_heat_cop / base_hp_heat_cop  # <1
+
+        hp_cop_scaler = pd.Series(dtype="float64")
+        for i in temp_df.index:
+            hp_cop_scaler.loc[i] = hp_cop_adv.loc[
+                round(temp_df.loc[i, "temp_c"], 1), "cop"
+            ]
+        hp_load_scaler = hp_cop_scaler * new_scen.frac_hp_growth(base_scen)
+
+    scen_load_mwh = pd.DataFrame(index=base_energy.index)
+    scen_load_mwh["base_load_mw"] = base_energy["base_load_mw"] * base_load_scaler
+    if (
+        new_hp_profile == "elec"
+    ):  # if user select to use the current electricity heat pump consumption profiles for heating load projection
+        scen_load_mwh["heat_hp_load_mw"] = (
+            base_energy["heat_hp_load_mw"] * hp_load_scaler
+        )
+    else:
+        scen_load_mwh["heat_existing_hp_load_mw"] = base_energy["heat_hp_load_mw"]
+    scen_load_mwh["heat_resist_load_mw"] = (
+        base_energy["heat_resist_load_mw"] * resist_load_scaler
+    )
+    scen_load_mwh["cool_load_mw"] = base_energy["cool_load_mw"] * cool_load_scaler
+
+    return scen_load_mwh
+
+
+def ff_electrify_profiles(weather_years, puma_data, new_scen, base_scen):
+    """return hourly electricity loads for a projection scenario from converting fossil fuel heating, dhw and cooking to electric ones
+    :param list weather_years: user defined year(s) of weather profile for load projection
+    :param pandas.DataFrame puma_data: puma data within zone, output of zone_shp_overlay()
+    :param class scenarios new_scen: projection scenario class
+    :param class scenarios base_scen: reference scenario class
```

Comment:
Instead of `class`, these input parameters are instances of such class. See the similar doc strings in the code base [here](https://github.com/Breakthrough-Energy/PostREISE/blob/c560aa6733e06e70b14faa573935282bd729ba9e/postreise/analyze/generation/capacity.py#L18)

Reply:
got it.